syntax = "proto3";

package redesocial;

// Mensagens para sincronização de relógio
message ClockRequest {
  double relogio_fisico = 1;
}

message ClockReply {
  double offset = 1;
}

// Mensagens principais
message Postagem {
  string user_id = 1;
  string conteudo = 2;
  double timestamp_logico = 3;
  double timestamp_fisico = 4;
}

message Mensagem {
  string from_ = 1;
  string to = 2;
  string conteudo = 3;
  double timestamp_logico = 4;
}

message SeguirRequest { // <<< RENOMEADO
  string seguidor_id = 1;
  string seguido_id = 2;
}

message Ack {
  string message = 1;
}

// Serviço
service RedeSocial {
  rpc Postar(Postagem) returns (Ack);
  rpc Seguir(SeguirRequest) returns (Ack); 
  rpc EnviarMensagem(Mensagem) returns (Ack);
  rpc SincronizarRelogio(ClockRequest) returns (ClockReply);
}
